src/crimson/osd/pg_backend.cc

```diff
@@ -833,12 +835,14 @@ PGBackend::rollback_iertr::future<> PGBackend::rollback(
     // if there's no snapshot, we delete the object;
     // otherwise, do nothing.
     crimson::ct_error::enoent::handle(
-    [this, &os, &snapid, &txn, &delta_stats] {
+    [this, &os, &snapid, &txn, &delta_stats, &snapc, &ss] {
       logger().debug("PGBackend::rollback: deleting head on {}"
                      " with snap_id of {}"
                      " because got ENOENT|whiteout on obc lookup",
                      os.oi.soid, snapid);
-      return remove(os, txn, delta_stats, false);
+      bool whiteout = !ss.clones.empty() ||
+	(!snapc.snaps.empty() && snapc.snaps[0] > ss.seq);
+      return remove(os, txn, delta_stats, whiteout);
```

Comment:
nit: Can this be moved to an helper method?
both CEPH_OSD_OP_DELETE and CEPH_OSD_OP_ROLLBACK re-use this logic.